### ISSUE TITLE
Infinite loop on any dispatch

### ISSUE THREAD

**Issue by gege83** [soyguijarro/react-storage-hooks#5]
First of all, i would like to thank that you have created this library.

We tried to use it with React ^16.9.0-alpha.0
we created a simple test component and with a state and a counter. When the button is pushed we dispatch an update to the counter. the screen keeps updating the counter. jumping between counter and counter+1. The session storage contains the right value.

code to reproduce:

```
import 'react-app-polyfill/ie11';
import 'ts-polyfill';

import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import * as serviceWorker from './serviceWorker';
import {useSessionStorageReducer} from 'react-storage-hooks';

function reducer(state, action) {
  return action;
}

function TestComponent() {
  const [state, dispatch] = useSessionStorageReducer('key', reducer, {some: 'value', counter: 0});
  return <button onClick={() => dispatch({...state, counter: state.counter + 1})}>{JSON.stringify(state)}</button>
}

ReactDOM.render(<TestComponent/>, document.getElementById('root'));

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: https://bit.ly/CRA-PWA
serviceWorker.unregister();
```

Thanks for your time and effort

**Comment by soyguijarro** [comment]
@gege83 Hey, thanks for reporting! Something is definitely wrong, I'll try to take a look in the next few days. In the meantime you can try using version 3.0.1, that one should work fine.

**Comment by timvanoostrom** [comment]
I have the same problem. 3.01 works as expected.

**Comment by soyguijarro** [comment]
@gege83 @timvanoostrom This should be fixed in version 4.0.0 that I've just published, so I'm closing this issue. Feel free to reopen if it keeps happening.
